operations/research-and-development/engineering/infrastructure-engineering/reliability-manifesto.md

```diff
@@ -0,0 +1,123 @@
+
+## Continuous Improvement
+
+There is always room for improvement in our knowledge and our systems. The continuous improvement manifesto aims to make us more efficient, resilient and better engineers.
+
+**C-1** We should track all our production incidents. Major and critical incidents are reported and updated using our statuspage. All incidents are tracked via a Infrastructure Incident Response v2.0 playbook run. More details here.
+
+**C-2** Post mortem meetings and documents should follow every major and critical production incident. Runbooks if any should be added in the internal documentation for future reference.
+
+**C-3** No single engineer should fully deploy a new service across all environments. We share knowledge and reduce the team bus factor. 
+
+**C-4** Blameless culture is a key. People make mistakes and systems (do) fail. We need to learn from them and help each other. 
+
+## Design & Architecture
+
+Better design and architectures create more reliable systems and reduce technical debt. This is what the Design & Architecture manifesto is all about. 
+
+**D-1** Infrastructure team should be involved early in the design process for applications that will run in the infrastructure systems. Read about the [Production Readiness Review](/operations/research-and-development/engineering/infrastructure-engineering/production-readiness-review.md)
+
+**D-2** All new services should go through the POC phase and findings should be presented in a team meeting, together with proposal, design architecture and implementation steps. Example reference [here](https://docs.google.com/document/d/1KZlemVB_75Xv4CDao5bcWpKml3fHbpcoAk_gWl8nFSY/edit).
+
+**D-3** We should avoid monolithic designs for custom services when possible. Microservice design should be the aim and services should be open sourced for community usage when they don’t expose private company information.
+ 
+**D-4** All services should be followed by architecture diagrams (we use Lucidchart) when possible and documentation. 
+
+**D-5** Always be mindful that services work in a different way at scale. Configurations that might work in local environments could have a major impact in production systems. For example, a possible abuse of the amount of database connections used. 
+
+**D-6** All services should support the ability to turn off/on on demand without affecting the rest of the infrastructure services.
+
+## Observability
+
+The observability manifesto should cover three main pillars:
+- Logs
+- Metrics
+- Tracing
+
+### Logs
+
+**O-1** Logs should include meaningful messages and stack trace should not be a requirement to understand the log message.
+
+**O-2** Service logs are enriched with metadata & attributes that differentiate the service (e.g service: boards).
+
+**O-3** Logging levels should be consistent across services and false errors should be avoided to enable generation of metrics, patterns and dashboards.
```

Comment:
```suggestion
**O-3** Logging levels should be consistent across services and false errors should be avoided to enable generation of metrics, patterns and dashboards. By default, applications will log at `INFO` and above and produce a relatively low volume of logs.
- Use `ERROR` when a human should take action to remediate a failure such as database connection issues, incorrect cluster configuration, inability to reach a first-party host (e.g. S3, rtcd) or an application segfault.
- Use `WARN` for something unexpected, but typically recoverable without human intervention, such as the inability to reach a third-party host (e.g. GitHub, Giphy), an unexpected payload, or a failed client API call. Assume that infrastructure tooling will monitor for a spike in warning logs to trigger human action as needed.
- Use `INFO` to capture context around non-user initiated events such as startup information, plugin state changes, schema migrations, background jobs starting and finishing, or configuration changes. 
- Use `DEBUG` to trace user-initiated events or add context to other systems such as client API requests, the details of the LDAP Sync job or cluster polling.
```

cc @agnivade for opinions on all this

Reply:
We should just point to: https://developers.mattermost.com/contribute/more-info/server/style-guide/#log-levels.

Reply:
@lieut-data let's add the link as proposal maybe then?

Reply:
Agreed, developer docs is good for me!